vars/ostreeBootSanity.groovy

```diff
@@ -0,0 +1,86 @@
+
+            // Set groovy and env vars
+            env.task = "./ci-pipeline/tasks/ostree-boot-image"
+            env.playbook = "ci-pipeline/playbooks/system-setup.yml"
+
+            // Provision resources
+            env.DUFFY_OP = "--allocate"
+            utils.duffyCciskel([stage:current_stage, duffyKey:'duffy-key', duffyOps:env.DUFFY_OP])
+
+            echo "Duffy Allocate ran for stage ${current_stage} with option ${env.DUFFY_OP}\r\n" +
+                 "ORIGIN_WORKSPACE=${env.ORIGIN_WORKSPACE}\r\n" +
+                 "ORIGIN_BUILD_TAG=${env.ORIGIN_BUILD_TAG}\r\n" +
+                 "ORIGIN_CLASS=${env.ORIGIN_CLASS}"
+
+            def job_props = "${env.ORIGIN_WORKSPACE}/job.props"
+            def job_props_groovy = utils.convertProps(job_props)
+            load(job_props_groovy)
+
+            // Stage resources - ostree boot sanity
+            pipelineUtils.setupStage(current_stage, 'fedora-atomic-key')
+
+            // Rsync Data - ostree boot sanity
+            writeFile file: "${env.ORIGIN_WORKSPACE}/task.env",
+                    text: "export branch=\"${branch}\"\n" +
+                          "export HTTP_BASE=\"${HTTP_BASE}\"\n" +
+                          "export RSYNC_USER=\"${RSYNC_USER}\"\n" +
+                          "export RSYNC_SERVER=\"${RSYNC_SERVER}\"\n" +
+                          "export RSYNC_DIR=\"${RSYNC_DIR}\"\n" +
+                          "export FEDORA_PRINCIPAL=\"${FEDORA_PRINCIPAL}\"\n" +
+                          "export fed_repo=\"${fed_repo}\"\n" +
+                          "export image2boot=\"${image2boot}\"\n" +
+                          "export commit=\"${commit}\"\n" +
+                          "export JENKINS_JOB_NAME=\"${JOB_NAME}-${current_stage}\"\n" +
+                          "export JENKINS_BUILD_TAG=\"${BUILD_TAG}-${current_stage}\"\n" +
+                          "export OSTREE_BRANCH=\"${OSTREE_BRANCH}\"\n" +
+                          "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+
+            pipelineUtils.rsyncResults(current_stage, 'duffy-key')
+
+            // Set Message Fields
+            (topic, messageProperties, messageContent) = pipelineUtils.setMessageFields('test.integration.queued')
+            env.topic = topic
+
+        }
+    } catch (err) {
+        echo "Error: Exception from " + current_stage + ":"
+        echo err.getMessage()
+        throw err
+    } finally {
+        // Teardown resources
+        env.DUFFY_OP = "--teardown"
+        echo "Duffy Deallocate ran for stage ${current_stage} with option ${env.DUFFY_OP}\r\n" +
+             "RSYNC_PASSWORD=${env.RSYNC_PASSWORD}\r\n" +
+             "DUFFY_HOST=${env.DUFFY_HOST}"
+        utils.duffyCciskel([stage:current_stage, duffyKey:'duffy-key', duffyOps:env.DUFFY_OP])
+
+        // Send message org.centos.prod.ci.pipeline.compose.test.integration.queued on fedmsg
+        messageUtils.sendMessage([topic:"${env.topic}",
```

Comment:
Should this be above where you get the values? If you throw an exception these variables will be undefined right?